hbase-assembly/pom.xml

```diff
@@ -245,20 +211,36 @@
       <groupId>io.opentelemetry.javaagent</groupId>
       <artifactId>opentelemetry-javaagent</artifactId>
     </dependency>
-    <!-- We don't really add this to assembly tarball, we retain it here just to dump it into
-    cached_classpath.txt ! See HBASE-28433 for more info. -->
+    <!-- This is an optional dependency of hbase-external-blockcache.
```

Comment:
If so, let's just remove the optional flag in hbase-external-blockcache's pom file?

Reply:
I took another look at this.

I guess that the original intent was to NOT include it in the assembly, it just got lost somehow.

The last version is also from 2017, so this doesn't look a maintaned library.

Based on the comments, getting this right is far from trivial, and this has few users, so it might be better to keep it optional, and just leave it out from assembly.

That way most user will get less JARs, and the few (if any) users of it would have to provide it themselves.

Reply:
Or we can do the removal in another ticket, so that it's better documented.

Reply:
I have opened a [DISCUSS] thread for spymemcached removal.